src/mbgl/shader/pattern.vertex.glsl

```diff
@@ -9,6 +12,6 @@ varying vec2 v_pos_b;
 
 void main() {
     gl_Position = u_matrix * vec4(a_pos, 0, 1);
-    v_pos_a = (u_patternmatrix_a * vec3(a_pos, 1)).xy;
-    v_pos_b = (u_patternmatrix_b * vec3(a_pos, 1)).xy;
+    v_pos_a = (u_patternmatrix_a * vec3(a_pos, 1)).xy + u_offset_a;
+    v_pos_b = (u_patternmatrix_b * vec3(a_pos, 1)).xy + u_offset_b;
```

Comment:
If removing the matrix like in -js sounds good, then it can be done separately from this pull request